user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/ValidationExecutor.java

```diff
@@ -0,0 +1,29 @@
+package life.qbic.datamanager.views.projects.project.measurements;
+
+import java.util.Collection;
+import java.util.Optional;
+import life.qbic.projectmanagement.application.measurement.NGSMeasurementMetadata;
+import life.qbic.projectmanagement.application.measurement.ProteomicsMeasurementMetadata;
+import life.qbic.projectmanagement.application.measurement.validation.ValidationResult;
+import life.qbic.projectmanagement.application.measurement.validation.ValidationService.Domain;
+
+/**
+ * <b>Validation Executor</b>
+ * <p>
+ * An interface that enables injection of different configurations of
+ * {@link life.qbic.projectmanagement.application.measurement.validation.ValidationService} calls,
+ * based on the use case (create or edit).
+ * <p>
+ * To support more domains, just create more methods here.
```

Comment:
This smells a bit. Why not have two different interfaces? Why not have a more general interface? Can you elaborate?
For example: `ValidationExecutor<T>` with `NGSValidationExecutor<NGSMEasurementMetadata>` and `ProteomicsValidationExecutor<ProteomicsMeasurementMetadata>` implementing 
`<T> ValidationResult validate(T metadata)`

Reply:
This is due to two reasons: 
1.) The dialog itself has to decide what kind of domain validation is necessary so we can't know during it's initialization what the domain will be.
2.) The idea of this interface was to seperate the edit and register cases for the different facilities because these cases have different validation needs
(Measurement Registration has no measurement codes, while editing measurements has to ensure the correct measurement codes were provided)

Reply:
to 
1.) Why? What is the reasoning to delegate that to the dialog?
2.) Still, I do not see a reason for different interfaces?

Reply:
To the provided questions
1.) The dialog will show the user immediately after uploading a file within the grey listbox of a file item the missing/wrong information for each uploaded file. 
Therefore the dialog has to know the domain of the uploaded file to trigger the correct validation. 

2.) The dialog has to know if the user wants to edit an existing measurement or add a new measurement, since: 
A) Some elements in the dialog will change (Save instead of register as the confirmbutton label e.g.). 

B) The provided information has to be validated differently dependent on if the user intends to edit or register a measurement. (Registrations will have no measurement codes,  User is not allowed to edit some of the provided columns in existing measurements. )